pkg/volume/csi/csi_attacher.go

```diff
@@ -64,88 +64,124 @@ func (c *csiAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (string
 		return "", errors.New("missing spec")
 	}
 
-	csiSource, err := getCSISourceFromSpec(spec)
+	var (
+		driverName   string
+		volumeHandle string
+		pvSource     *v1.CSIPersistentVolumeSource
+		node         = string(nodeName)
+	)
+	volSource, pvSource, err := getSourceFromSpec(spec)
 	if err != nil {
 		klog.Error(log("attacher.Attach failed to get CSI persistent source: %v", err))
 		return "", err
 	}
 
-	skip, err := c.plugin.skipAttach(csiSource.Driver)
+	if c.plugin.inlineFeatureEnabled && volSource != nil {
+		driverName = volSource.Driver
+		if volSource.VolumeHandle != nil {
+			volumeHandle = *volSource.VolumeHandle
+		} else {
+			return "", errors.New("CSI volume source missing required volumeHandle")
+		}
+	} else if pvSource != nil {
+		driverName = pvSource.Driver
+		skip, err := c.plugin.skipAttach(driverName)
+		if err != nil {
+			klog.Error(log("attacher.Attach failed to find if driver is attachable: %v", err))
+			return "", err
+		}
+		if skip {
+			klog.V(4).Infof(log("skipping attach for driver %s", driverName))
+			return "", nil
+		}
+		volumeHandle = pvSource.VolumeHandle
+	} else {
+		return "", errors.New("spec missing CSI volume source")
+	}
+
+	attachID := getAttachmentName(volumeHandle, driverName, node)
+	attachment, err := c.newVolumeAttachment(attachID, node, driverName, spec)
 	if err != nil {
-		klog.Error(log("attacher.Attach failed to find if driver is attachable: %v", err))
+		klog.Error(log("attacher.Attach unable to create VolumeAttachment object: %v", err))
 		return "", err
 	}
-	if skip {
-		klog.V(4).Infof(log("skipping attach for driver %s", csiSource.Driver))
-		return "", nil
+
+	// post attachment object and wait for attachment to be created
+	createdAttchID, err := c.postVolumeAttachment(driverName, volumeHandle, attachment, csiDefaultTimeout)
```

Comment:
`post` is an overloaded word and means something fairly specific. Is that what we're doing? Or is this more of a CreateVolumeAttachment type deal

Reply:
Yeah, I think this is a create volume attachment thing. post as in http method=post.